src/test/java/org/benetech/servicenet/service/impl/ImportServiceConfidentialityImplIntTest.java

```diff
@@ -295,213 +333,159 @@ public void shouldNotUpdateServiceIfConfidential() {
 
-        languageService.findAll().size();
-        importService.createOrUpdateLangsForService(Set.of(language), service, null);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
         assertEquals(0, languageService.findAll().size());
     }
 
     @Test
     @Transactional
     public void shouldNotReplaceLangsIfConfidentialEvenIfServiceHasFewOfThemButNotThisOne() {
-        Service service = generateExistingService();
-        Language otherLanguage = new Language().language(OTHER_STRING).srvc(service);
-        em.persist(otherLanguage);
-        em.flush();
-        em.refresh(service);
-
-        Language newLanguage = new Language().language(NEW_STRING).srvc(service);
+        Language otherLanguage = new Language().language(OTHER_STRING);
+        Language newLanguage = new Language().language(NEW_STRING);
         newLanguage.setIsConfidential(true);
+        Service service = generateExistingService().langs(Set.of(otherLanguage, newLanguage));
 
-        languageService.findAll().size();
-        importService.createOrUpdateLangsForService(Set.of(newLanguage), service, null);
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(0, languageService.findAll().size());
+        assertEquals(1, languageService.findAll().size());
+        assertEquals(OTHER_STRING, languageService.findAll().get(0).getLanguage());
     }
 
     @Test
     @Transactional
     public void shouldNotReplaceLangsIfConfidential() {
-        Service service = generateExistingService();
-
-        Language languageToBeUpdated = new Language().language(EXISTING_STRING).srvc(service);
-        em.persist(languageToBeUpdated);
-
-        Language otherLanguage = new Language().language(OTHER_STRING).srvc(service);
-        em.persist(otherLanguage);
-        em.flush();
-        em.refresh(service);
+        Language languageToBeUpdated = new Language().language(EXISTING_STRING);
+        Service service = generateExistingService().langs(Set.of(languageToBeUpdated));
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        Language newLanguage = new Language().language(NEW_STRING).srvc(service);
-        newLanguage.setIsConfidential(true);
-
-        languageService.findAll().size();
-        importService.createOrUpdateLangsForService(Set.of(newLanguage), service, null);
+        Language otherLanguage = new Language().language(OTHER_STRING);
+        otherLanguage.setIsConfidential(true);
+        service.setLangs(Set.of(otherLanguage));
+        importManager.createOrUpdateService(service, EXISTING_EXTERNAL_ID, PROVIDER, new DataImportReport());
 
-        assertEquals(0, languageService.findAll().size());
+        assertEquals(1, languageService.findAll().size());
+        assertEquals(EXISTING_STRING, languageService.findAll().get(0).getLanguage());
```

Comment:
In my opinion, this is misleading because the lang is outdated.

Reply:
You're right in this case, it shouldn't happened.